### PR TITLE
Update the JSDoc for enableIndexedDbPersistence

### PR DESCRIPTION
**Pull request by milaGGL**
In some situations, `enableIndexedDbPersistence` and `enableMultiTabIndexedDbPersistence` throw an error instead of rejecting a promise. Rejected promise is documented as the way the error will be surfaced.
https://github.com/firebase/firebase-js-sdk/blob/03ba26296ef029c844f9fbd6c11e3c773cf8eb14/packages/firestore/test/integration/browser/indexeddb.test.ts#L41

https://github.com/firebase/firebase-js-sdk/blob/03ba26296ef029c844f9fbd6c11e3c773cf8eb14/packages/firestore/test/integration/api/provider.test.ts#L151
Updated the documentation accordingly.

### CODE REVIEW

**Review by dconeybe — CHANGES_REQUESTED** [review]
Looking at the code, I don't think the documentation updates are correct. It appears that both the `enableIndexedDbPersistence()` and `enableMultiTabIndexedDbPersistence()`  functions directly throw exceptions for some problems, and return a promise that is ultimately rejected for other problems.

Specifically, it looks like they throw an exception when misused and return a promise that eventually gets rejected if some problem out of the programmer's control prevents persistence from being set (e.g. indexeddb is not available).

So I'd recommend to revert your changes in this PR, then, just _add_ some information, to indicate that an exception is thrown if persistence has already been enabled or if it is too late to enable persistence. Of course, verify that this is indeed correct. I'm just concerned that the updated documentation is still not fully correct.

File: packages/firestore/src/api/database.ts [hunk 1]
```diff
@@ -305,20 +305,21 @@ export function configureFirestore(firestore: Firestore): void {
 /**
  * Attempts to enable persistent storage, if possible.
  *
- * Must be called before any other functions (other than
- * {@link initializeFirestore}, {@link (getFirestore:1)} or
- * {@link clearIndexedDbPersistence}.
+ * If fails, `enableMultiTabIndexedDbPersistence()` will reject the promise or
```
Comment (dconeybe) [inline packages/firestore/src/api/database.ts:308]: nit: "enableMultiTabIndexedDbPersistence" -> "enableIndexedDbPersistence".

Comment (dconeybe) [inline packages/firestore/src/api/database.ts:308]: nit: "If fails" -> "On failure"

File: packages/firestore/src/api/database.ts [hunk 2]
```diff
@@ -305,20 +305,21 @@ export function configureFirestore(firestore: Firestore): void {
...
+ * throw an error. There are several reasons why this can fail, which can be
```
Comment (dconeybe) [inline packages/firestore/src/api/database.ts:309]: nit: "throw an error" -> "throw an exception"

File: packages/firestore/src/api/database.ts [hunk 3]
```diff
@@ -305,20 +305,21 @@ export function configureFirestore(firestore: Firestore): void {
...
- * If this fails, `enableIndexedDbPersistence()` will reject the promise it
- * returns. Note that even after this failure, the {@link Firestore} instance will
- * remain usable, however offline persistence will be disabled.
+ *   * failed-precondition: The app is already open in another browser tab.
+ *   * unimplemented: The browser is incompatible with the offline persistence
+ *     implementation.
  *
- * There are several reasons why this can fail, which can be identified by
- * the `code` on the error.
+ * Note that even after this failure, the {@link Firestore} instance will remain
```
Comment (dconeybe) [inline packages/firestore/src/api/database.ts:316]: nit: "even after this failure" -> "even after a failure"

File: packages/firestore/src/api/database.ts [hunk 4]
```diff
@@ -305,20 +305,21 @@ export function configureFirestore(firestore: Firestore): void {
...
+ * usable, however offline persistence will be disabled.
  *
- *   * failed-precondition: The app is already open in another browser tab.
- *   * unimplemented: The browser is incompatible with the offline
- *     persistence implementation.
+ * Note: `enableIndexedDbPersistence()` must be called before any other functions
+ * (other than {@link initializeFirestore}, {@link (getFirestore:1)} or
+ * {@link clearIndexedDbPersistence}. Failing to do so will make this
```
Comment (dconeybe) [inline packages/firestore/src/api/database.ts:321]: I would omit the sentence beginning with "Failing to do so will..."

File: packages/firestore/src/api/database.ts [hunk 5]
```diff
@@ -371,17 +372,17 @@ export function enableIndexedDbPersistence(
  * shared execution of queries and latency-compensated local document updates
  * across all connected instances.
  *
- * If this fails, `enableMultiTabIndexedDbPersistence()` will reject the promise
- * it returns. Note that even after this failure, the {@link Firestore} instance will
- * remain usable, however offline persistence will be disabled.
- *
- * There are several reasons why this can fail, which can be identified by
- * the `code` on the error.
+ * If fails, `enableMultiTabIndexedDbPersistence()` will reject the promise or
```
Comment (dconeybe) [inline packages/firestore/src/api/database.ts:375]: Please make the same changes to the docs for `enableMultiTabIndexedDbPersistence()` as are made to `enableIndexedDbPersistence()`.

**Review by dconeybe — CHANGES_REQUESTED** [review]
A few minor nits. Please take my comments as suggestions, rather than orders.

**Review by dconeybe — APPROVED** [review]

**Review by markarndt — APPROVED** [review]

**Pull request merged**